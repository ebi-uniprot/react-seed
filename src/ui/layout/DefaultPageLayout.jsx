
import React from 'react';
import PropTypes from 'prop-types';

import DefaultPageContent from './DefaultPageContent';

const DefaultPageLayout = props => (
  <div className="default-page-layout">

    You are viwing <b>{props.title}</b>
    <br />

    <DefaultPageContent>
      {props.content}
    </DefaultPageContent>
  </div>
);

DefaultPageLayout.propTypes = {
  title: PropTypes.string,
  content: PropTypes.element,
};

DefaultPageLayout.defaultProps = {
  title: 'Page Title',
  content: () => <h3>Page Content</h3>,
};

export default DefaultPageLayout;
