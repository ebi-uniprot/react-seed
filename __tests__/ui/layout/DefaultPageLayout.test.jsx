
import React from 'react';
import renderer from 'react-test-renderer';

import DefaultPageLayout from '../../../src/ui/layout/DefaultPageLayout';

describe('DefaultPageLayout component', () => {
  test('should render', () => {
    let component = (
      <DefaultPageLayout
        title="Works"
        content={<h3>Works</h3>}
      />);

    component = renderer
      .create(component)
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
