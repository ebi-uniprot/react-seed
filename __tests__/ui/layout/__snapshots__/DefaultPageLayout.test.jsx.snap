// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultPageLayout component should render 1`] = `
<div
  className="default-page-layout"
>
  You are viewing 
  <b>
    Works
  </b>
  <br />
  <div
    className="default-page-conent"
  >
    <h3>
      Works
    </h3>
  </div>
</div>
`;
